tests/test_helpers.py

```diff
@@ -34,240 +34,230 @@ def has_encoding(name):
 
-    def test_post_empty_json_wont_add_exception_to_response_if_no_debug(self):
-        app = flask.Flask(__name__)
+    def test_post_empty_json_wont_add_exception_to_response_if_no_debug(self, app, client):
         app.config['DEBUG'] = False
+
         @app.route('/json', methods=['POST'])
         def post_json():
             flask.request.get_json()
             return None
-        c = app.test_client()
-        rv = c.post('/json', data=None, content_type='application/json')
+
+        rv = client.post('/json', data=None, content_type='application/json')
         assert rv.status_code == 400
         assert b'Failed to decode JSON object' not in rv.data
 
-    def test_json_bad_requests(self):
-        app = flask.Flask(__name__)
+    def test_json_bad_requests(self, app, client):
+
         @app.route('/json', methods=['POST'])
         def return_json():
             return flask.jsonify(foo=text_type(flask.request.get_json()))
-        c = app.test_client()
-        rv = c.post('/json', data='malformed', content_type='application/json')
+
+        rv = client.post('/json', data='malformed', content_type='application/json')
         assert rv.status_code == 400
 
-    def test_json_custom_mimetypes(self):
-        app = flask.Flask(__name__)
+    def test_json_custom_mimetypes(self, app, client):
+
         @app.route('/json', methods=['POST'])
         def return_json():
             return flask.request.get_json()
-        c = app.test_client()
-        rv = c.post('/json', data='"foo"', content_type='application/x+json')
+
+        rv = client.post('/json', data='"foo"', content_type='application/x+json')
         assert rv.data == b'foo'
 
-    def test_json_body_encoding(self):
-        app = flask.Flask(__name__)
+    def test_json_body_encoding(self, app, client):
         app.testing = True
+
         @app.route('/')
         def index():
             return flask.request.get_json()
 
-        c = app.test_client()
-        resp = c.get('/', data=u'"Hällo Wörld"'.encode('iso-8859-15'),
-                     content_type='application/json; charset=iso-8859-15')
+        resp = client.get('/', data=u'"Hällo Wörld"'.encode('iso-8859-15'),
+                          content_type='application/json; charset=iso-8859-15')
         assert resp.data == u'Hällo Wörld'.encode('utf-8')
 
-    def test_json_as_unicode(self):
-        app = flask.Flask(__name__)
+    def test_json_as_unicode(self, app, app_ctx):
 
         app.config['JSON_AS_ASCII'] = True
-        with app.app_context():
-            rv = flask.json.dumps(u'\N{SNOWMAN}')
-            assert rv == '"\\u2603"'
+        rv = flask.json.dumps(u'\N{SNOWMAN}')
+        assert rv == '"\\u2603"'
 
         app.config['JSON_AS_ASCII'] = False
-        with app.app_context():
-            rv = flask.json.dumps(u'\N{SNOWMAN}')
-            assert rv == u'"\u2603"'
+        rv = flask.json.dumps(u'\N{SNOWMAN}')
+        assert rv == u'"\u2603"'
```

Comment:
Note that your refactoring here runs both tests within a single app_context as the fixture teardown isn't executed until you exist the test function. Versus the previous way used separate app contexts. I don't think it matters in this case, although I'm uncertain as it's been a while since I've looked at what `app_context()` is actually doing.

You might consider refactoring this using `pytest` parameterization. Pass the test function `True`/`False` for `JSON_AS_ASCII` as well as the expected output. This would also stay within the original behavior of creating a new app context each run through.